### DISCUSSION TITLE
antdv5版本开启cssVar配置之后，如何动态的修改这些css变量呢

### DISCUSSION THREAD

**Discussion by york-chen** [ant-design/ant-design#48185] [Q&A]
我使用的umi架构，然后我在配置里面开启了css变量
antd: {
    theme:{
      cssVar:{
        key: 'body1',
        prefix:''
      },
      hashed:false,
    },
  },
但是这样在html结构里面去看，这些css变量被声明在了 .body1这个类下面，如果我想动态的去修改这些变量，我得document.querySelectorAll('.body1111').forEach(e=>e.style?.setProperty('--color-link', 'red'));
我感觉这样的写法不是很优雅呢，因为页面上会生成好几个div，他的class包含了 body1
所以我想请教一下，既然antd 暴露了css变量这个配置，那我应该如何优雅的使用它，并且动态的切换主题呢

**Comment by YuShai** [comment]
document.documentElement.style.setProperty('--primary-color', '#ff0000');

**Reply by york-chen** [reply]
变量没有定义在根节点，直接这样设置不会生效的吧
